src/test_util.rs

```diff
@@ -69,6 +70,48 @@ pub fn send_lines(
         })
 }
 
+pub fn send_lines_tls(
+    addr: SocketAddr,
+    host: String,
+    lines: impl Iterator<Item = String>,
+) -> impl Future<Item = (), Error = ()> {
+    let lines = futures::stream::iter_ok::<_, ()>(lines);
+
+    let connector: TlsConnector = native_tls::TlsConnector::builder()
+        .danger_accept_invalid_certs(true)
+        .danger_accept_invalid_hostnames(true)
+        .build()
+        .expect("Failed to build TLS connector")
+        .into();
+
+    TcpStream::connect(&addr)
+        .map_err(|e| panic!("{:}", e))
+        .and_then(move |socket| {
+            connector
+                .connect(&host, socket)
+                .map_err(|e| panic!("{:}", e))
+                .and_then(|stream| {
+                    let out = FramedWrite::new(stream, LinesCodec::new())
+                        .sink_map_err(|e| panic!("{:?}", e));
+
+                    lines
+                        .forward(out)
+                        .and_then(|(_source, sink)| {
+                            let mut stream = sink.into_inner().into_inner();
+                            // We should catch TLS shutdown errors here,
```

Comment:
I am not directly following why we need to handle the shutdown here? the forward future should in theory flush all the way?

Reply:
The parallel `send_lines` function has a `shutdown`, so I tried to get it to work in the TLS version and was unable. I am uncomfortable removing one without the other.

Reply:
Yeah, so I think dropping the tcpstream should be fine enough since forward should ensure its all flushed. So I think its safe to ignore it.

Reply:
The tests that use this new `send_lines_tls` have passed every time I tried, so if there is a race or other issue with this it isn't obvious. However, I left the comment there because of the discrepancy.